### PR TITLE
Insecure connection handling

### PR DESCRIPTION
**Pull request by surya9839**

### CODE REVIEW

File: k8tls-job/templates/k8tls-job.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ spec:
       containers:
       - image: accuknox/accuknox-job:latest
         command: ["/bin/sh", "-c"]
-        args: ['curl --location --request POST "https://${URL}/api/v1/artifact/?tenant_id=${TENANT_ID}&data_type=K8TLS&save_to_s3=false" --header "Tenant-Id: ${TENANT_ID}" --header "Authorization: Bearer ${AUTH_TOKEN}" --form "file=@\"/data/report.json\"" && cat /data/report.json']
+        args: ['./curl_command.sh']
```
Comment (nyrahul) [inline k8tls-job/templates/k8tls-job.yaml:15]: data_type is hardcoded to KB in the curl_command.sh script.

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: k8tls-job/templates/k8tls-job.yaml (hunk 1 above)
Comment (achrefbensaad) [reply k8tls-job/templates/k8tls-job.yaml:15]:
```suggestion
curl --location --request POST "https://${URL}/api/v1/artifact/?tenant_id=${TENANT_ID}&data_type=${DATA_TYPE}&label_id=${LABEL_NAME}&save_to_s3=true" \
```
you can pass the type as variable

**Review by achrefbensaad — CHANGES_REQUESTED** [review]
Facing this issue when running the helm chart
image

**Review by achrefbensaad — CHANGES_REQUESTED** [review]
Squash all commits and resolve the conflict

**Comment by surya9839** [comment]
@achrefbensaad there was issue with the image as changes were not pushed with this tag ,no changes are required in repo.
i have updated image with the same tag .

**Review by SujithKasireddy — APPROVED** [review]

File: curl_command.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    fi
+fi
+
+# main curl command
+curl --location --request POST "https://${URL}/api/v1/artifact/?tenant_id=${TENANT_ID}&data_type=${DATA_TYPE}&label_id=${LABEL_NAME}&save_to_s3=true" \
+    --header "Tenant-Id: ${TENANT_ID}" \
+    --header "Authorization: Bearer ${AUTH_TOKEN}" \
+    $CURL_FLAGS \
+    --form "file=@/data/report.json"
+
```
Comment (achrefbensaad) [inline curl_command.sh:31]: handle  the case where the curl return a non zero value

```suggestion
curl --location --request POST "https://${URL}/api/v1/artifact/?tenant_id=${TENANT_ID}&data_type=${DATA_TYPE}&label_id=${LABEL_NAME}&save_to_s3=true" \
    --header "Tenant-Id: ${TENANT_ID}" \
    --header "Authorization: Bearer ${AUTH_TOKEN}" \
    $CURL_FLAGS \
    --form "file=@/data/report.json" || exit 1

```

**Review by achrefbensaad — CHANGES_REQUESTED** [review]

**Review by achrefbensaad — APPROVED** [review]

**Comment by achrefbensaad** [comment]
Please squash before merging

File: .DS_Store
Comment (DelusionalOptimist) [inline .DS_Store:1]: Remove binary file

**Comment by DelusionalOptimist** [comment]
PR reverted. PTAL at the added comments and recreate after testing on local.
cc @SujithKasireddy @surya9839

**Pull request merged**